### PR TITLE
melkehitys-1020

### PR DESCRIPTION
**Pull request by natlibfi-jonollil**
removed blob transformation queue cleaning form dispatch and added it to cleanup as job

### CODE REVIEW

File: src/jobs/cleanup.js [hunk 1]
```diff
@@ -106,10 +171,27 @@ export default function (agenda) {
...
+								label: [
+									'fi.nationallibrary.melinda.record-import.container-type=transform-task',
+									`blobId=${id}`
+								]
+							}
+						});
+
+						if (containers.length === 0) {
+							logger.log('warn', `Blob ${id} has no transformer alive. Setting state to PENDING_TRANSFORMATION`);
+							client.updateState({id, state: BLOB_STATE.PENDING_TRANSFORMATION});
```
Comment (natlibfi-arlehiko) [inline src/jobs/cleanup.js]: The promise needs to be returned here. Otherwise will proceed to next client call below.

File: src/jobs/cleanup.js [hunk 2]
```diff
@@ -85,10 +146,14 @@ export default function (agenda) {
 
 			await processBlobs({
 				client, processCallback, messageCallback,
-				query: {state: [BLOB_STATE.PROCESSED, BLOB_STATE.ABORTED]},
+				query: {state},
 				filter: blob => {
 					const modificationTime = moment(blob.modificationTime);
-					return modificationTime.add(ttl).isBefore(moment());
+					if (before) {
```
Comment (natlibfi-arlehiko) [inline src/jobs/cleanup.js]: No need to introduce a new option because the `method`-option can be used here as well: `method === 'reQueueBlob`

**Comment by natlibfi-jonollil** [comment]
package update tested locally

File: src/jobs/dispatch.js [hunk 3]
```diff
@@ -299,34 +296,8 @@ export default function (agenda) {
 		}
 
 		async function processCallback(blobs) {
```
Comment (natlibfi-arlehiko) [inline src/jobs/dispatch.js]: Implementation is missing.

**Pull request merged**